statsmodels/tsa/tests/test_arima_process.py

```diff
@@ -41,6 +41,25 @@ def test_arma_acovf():
     rep2 = [1. * sigma * phi ** i / (1 - phi ** 2) for i in range(N)]
     assert_almost_equal(rep1, rep2, 7)  # 7 is max precision here
 
+def test_arma_acovf_persistent():
+    # Test arma_acovf in case where there is a near-unit root.
+    # .999 is high enough to trigger the "while ir[-1] > 5*1e-5:" clause,
+    # but not high enough to trigger the "nobs_ir > 50000" clause.
+    ar = np.array([1, -.9995])
+    ma = np.array([1])
+    process = ArmaProcess(ar, ma)
+    res = process.acovf(10)
+
+    # Theoretical variance sig2 given by:
+    # sig2 = .9995**2 * sig2 + 1
+    sig2 = 1/(1-.9995**2)
+
+    corrs = np.array([.9995**n for n in range(10)])
+    expected = sig2*corrs
+    assert len(res) == 10, len(res)
+    assert np.ndim(res) == 1
+    assert_almost_equal(res, expected, 6)
```

Comment:
assert_allclose is now preferred over assert_almost_equal because it has more option and allows specifying atol and rtol.
(assert_almost_equal existed in numpy for much longer, but is more like comparing to a printed table with rounding to fixed decimal.)